internal/controller/k8s_resources.go

```diff
@@ -85,9 +85,34 @@ func convertMetaObjectToMetadata(ctx context.Context, logger *zap.SugaredLogger,
 		}
 		objMetadata.KindSpecific = &pb.KubernetesObjectData_Pod{Pod: &pb.KubernetesPodData{IpAddresses: convertHostIPsToStrings(hostIPs)}}
 	}
+	if resource == "nodes" {
+		providerId, err := getProviderIdNodeSpec(ctx, logger, obj.GetName())
+
+		if err != nil {
+			return objMetadata, nil
+		}
+		objMetadata.KindSpecific = &pb.KubernetesObjectData_Node{Node: &pb.KubernetesNodeData{ProviderId: providerId}}
+	}
 	return objMetadata, nil
 }
 
+// getProviderIdNodeSpec uses a node name to grab the providerId within the nodeSpec
+func getProviderIdNodeSpec(ctx context.Context, logger *zap.SugaredLogger, nodeName string) (string, error) {
+	clientset, err := NewClientSet()
+	if err != nil {
+		logger.Errorw("Failed to create clientset", "error", err)
+		return "", err
+	}
+	node, err := clientset.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	if err != nil {
+		return "", nil
+	}
+	if node.Spec.ProviderID != "" {
+		return node.Spec.ProviderID, nil
+	}
+	return "", nil
```

Comment:
The semantics in the caller is that it sends a `KubernetesNodeData` if the returned error is `nil`, and doesn't when the error is not `nil`. Stick to that contract: if you have no provider ID to return, return a non-nil error.
```suggestion
	return "", errors.New("no providerID set")
```
Alternatively, modify the caller to ignore the returned provider ID if it is `""`.